src/pcre2_compile.c

```diff
@@ -7527,9 +7518,9 @@ for (;; pptr++)
       here because it just makes it horribly messy. */
 
       default:
-      if (op_previous >= OP_EODN)   /* Not a character type - internal error */
+      if (op_previous >= OP_EODN || op_previous <= OP_WORD_BOUNDARY)
         {
-        *errorcodeptr = ERR10;
+        *errorcodeptr = ERR10;  /* Not a character type - internal error */
```

Comment:
shouldn't there be a `PCRE2_DENUG_UNREACHABLE()` here?

Reply:
Good idea. There are quite a few places where we return an "internal error" but don't have an assertion. I'll hunt for a few and add some.

Reply:
actually, I meant to add it AFTER setting the errorccode, so the core dump has the correct value ;)

Reply:
Does it matter? The line number will be printed in the dump, which is what matters.

I'll change it if you like though.

Reply:
(I have a weak preference for putting the "unreachable" assertion at the _start_ of an unreachable, for clarity and consistency, so that line coverage can more easily exclude all the fallback code as unreachable.)

Reply:
Great, thanks for discussing!